simplq/src/components/pages/CreateQueue.jsx

```diff
@@ -11,76 +11,66 @@ import { SimplQHeader } from '../common/Header';
 import { handleEnterPress, isQueueNameValid } from '../common/utilFns';
 import InputField from '../common/InputField';
 
-class CreateQueue extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      textFieldValue: '',
-      invalidMsg: '', // state variable to display reason for invalid queue name
-      createInProgress: false,
-    };
-  }
+const CreateQueue = ({ history }) => {
+  const [textFieldValue, setTextFieldValue] = useState('');
+  const [invalidMsg, setInvalidMsg] = useState('');
+  const [createInProgress, setCreateInProgress] = useState(false);
 
-  handleClick = (queueName) => {
-    if (this.state.textFieldValue === '') this.setState({ invalidMsg: 'Queue name is required' });
+  const handleClick = (queueName) => {
+    if (textFieldValue === '') setTextFieldValue('Queue name is required');
     else {
-      this.setState({ createInProgress: true });
+      setCreateInProgress(true);
       QueueService.create(queueName)
         .then((response) => {
           store.dispatch(setQueueId(response.queueId));
           store.dispatch(setQueueName(response.queueName));
           store.dispatch(setCreationStep(1));
-          this.props.history.push('/admin');
+          history.push('/admin');
         })
```

Comment:
This would conflict with https://github.com/SimplQ/simplQ-frontend/pull/160

Reply:
Could you change it to history.push(`/queue/{response.queueId}`) and remove the store.dispatch(setQueueId(response.queueId));

Reply:
Best to wait till #160 is merged before this change though, to atleast test and make sure we're not missing aything else

Reply:
yep. will merge post #160

Reply:
#160 is merged